src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -95,56 +101,76 @@ public FileStream(SafeFileHandle handle, FileAccess access, int bufferSize, bool
         {
             ValidateHandle(handle, access, bufferSize, isAsync);
 
-            _strategy = WrapIfDerivedType(FileStreamHelpers.ChooseStrategy(handle, access, bufferSize, isAsync));
+            _strategy = FileStreamHelpers.ChooseStrategy(this, handle, access, bufferSize, isAsync);
         }
 
-        public FileStream(string path, FileMode mode) :
-            this(path, mode, mode == FileMode.Append ? FileAccess.Write : FileAccess.ReadWrite, DefaultShare, DefaultBufferSize, DefaultIsAsync)
-        { }
+        public FileStream(string path, FileMode mode)
+            : this(path, mode, mode == FileMode.Append ? FileAccess.Write : FileAccess.ReadWrite, DefaultShare, DefaultBufferSize, DefaultIsAsync)
+        {
+        }
```

Comment:
I actually prefer the current style since it wastes less space... I assume you are following rule 1 from [coding guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md)?

Reply:
> I assume you are following rule 1 from

Yes, I also wanted to be consistent with other `FileStream` `ctors`:

https://github.com/dotnet/runtime/blob/122c438e0987bf7bfd2a4a02368353d3774ff963/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs#L23-L39